wgpu-core/src/device/resource.rs

```diff
@@ -1368,6 +1369,11 @@ impl<A: HalApi> Device<A> {
             .iter(self_id.backend())
             .find(|&(_, bgl)| bgl.device_id.value.0 == self_id && bgl.entries == *entry_map)
             .map(|(id, value)| {
+                if let Some(comaptible) = value.compatible_layout {
+                    guard[comaptible].multi_ref_count.inc();
+                    return comaptible.0;
+                }
+
```

Comment:
I think we should instead filter out BGL's if they have a `compatible_layout`. (making it an invariant that the chain of BGL's can be at most 2; or put another way, the referenced `compatible_layout` can't itself have a `compatible_layout`).

Reply:
You are right I, made the change and it looks simpler.